### PR TITLE
LPS-178345 Allow developers to manage categories for object entries

### PR DESCRIPTION
**Pull request by 4lejandrito**
Adding support for taxonomy categories in objects.

To test it use the automated tests, or the API explorer.

**Important:** I've used the same `if` approach to handle the `taxonomyCategoryIds` field, as in `keywords`. I have created https://issues.liferay.com/browse/LPS-178772 to unify in a common API that allows us to get rid of the `ifs`. I am starting to work on it now, but I want this to get reviewed and merged first.

### CODE REVIEW

**Comment by 4lejandrito** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
ci:test:relevant

**Comment by 4lejandrito** [comment]
[Test failure](https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/399086//testReport/com.liferay.object.rest.internal.resource.v1_0.test/ObjectEntryResourceTest/testGetObjectEntryWithTaxonomyCategoriesAndEmbeddedTaxonomyCategory) is related. Weird because it passes locally. I'll investigate.

**Comment by LuismiBarcos** [comment]
I'm having the same problems with other PRs with the tests. I think it is because of the FF, but I am not sure.

**Review by LuismiBarcos — APPROVED** [review]
LGTM :)
Just added some irrelevant comments 😅

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java [hunk 1]
```diff
@@ -92,6 +99,21 @@ public static void setUpClass() throws Exception {
...
+		TaxonomyCategoryResource.Builder builder =
+			TaxonomyCategoryResource.builder();
+
+		_taxonomyCategoryResource = builder.authentication(
+			"test@liferay.com", "test"
+		).locale(
+			LocaleUtil.getDefault()
+		).build();
+
+		_assetVocabulary = AssetVocabularyLocalServiceUtil.addVocabulary(
```
[the remark this answers is not in the document]
Comment (4lejandrito) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java:113]: yes, I just need the same one for all the tests.

File: modules/apps/object/object-rest-test/build.gradle [hunk 2]
```diff
@@ -6,6 +6,7 @@ dependencies {
 	testIntegrationCompile group: "org.apache.cxf", name: "cxf-core", version: "3.4.10"
 	testIntegrationCompile project(":apps:account:account-api")
 	testIntegrationCompile project(":apps:document-library:document-library-api")
+	testIntegrationCompile project(":apps:headless:headless-admin-taxonomy:headless-admin-taxonomy-client")
```
[the remark this answers is not in the document]
Comment (4lejandrito) [reply modules/apps/object/object-rest-test/build.gradle:9]: This is to create categories using the headless apis. I could use the services instead. Do you think it is too bad?

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
ci:forward

**Comment by 4lejandrito** [comment]
https://github.com/brianchandotcom/liferay-portal/pull/132215

**Pull request closed without merging**